src/plot_api/plot_api.js

```diff
@@ -1701,15 +1701,26 @@ exports.relayout = function relayout(gd, astr, val) {
         seq.push(subroutines.layoutReplot);
     }
     else if(Object.keys(aobj).length) {
-        Plots.supplyDefaults(gd);
+        var flagList = Object.keys(flags).filter(function(k) { return flags[k]; });
+
+        // Optimization mostly for large splom traces where
+        // Plots.supplyDefaults can take > 100ms
+        if(flagList[0] === 'axrange' && flagList.length === 1) {
```

Comment:
Interesting... this seems safe to me, I'm mostly frustrated that we're locking ourselves into the "older" `relayout` for performance here, `react` won't be able to match it. Not to say we shouldn't do this - this is a fantastic optimization! But it might be worthwhile looking carefully at `supplyDefaults`, if there are any perf gains to be had there it would benefit everybody. Not in this PR of course...

Reply:
> But it might be worthwhile looking carefully at supplyDefaults, if there are any perf gains to be had there it would benefit everybody. Not in this PR of course...

Yeah for sure. https://github.com/plotly/plotly.js/issues/2549 will remain open after this PR.

Reply:
Added supply-default bypass to `Plotly.update` in https://github.com/plotly/plotly.js/pull/3057/commits/6de1ae4f1da1f5d84fd5c8d95d27f688132e2b17